pkg/internal/imetrics/iprom.go

```diff
@@ -100,3 +106,11 @@ func (p *PrometheusReporter) OTELTraceExportError(err error) {
 func (p *PrometheusReporter) PrometheusRequest(port, path string) {
 	p.prometheusRequests.WithLabelValues(port, path).Inc()
 }
+
+func (p *PrometheusReporter) DiscoverService(service string) {
+	p.discoveredServices.WithLabelValues(service).Set(1)
```

Comment:
That could be a bit confusing if, e.g. there are 4 python services that are reported generically as "python".

The user would se always `1`. Also, when a given process is stopped but other instances are still running, they would be set to zero.

Is it possible to keep internally how many instances of a given processes are there?

Reply:
That sounds like a good idea. I can change it to a Counter, but once we have more granularity on what we instrument (i.e which python programm is running) a Gauge would be more ideal for this.

Reply:
Nvm, I can use a Gauge here too